### ISSUE TITLE
Write UTxO-HD overview md docs

### ISSUE THREAD

**Issue by jasagredo** [IntersectMBO/ouroboros-consensus#1337]
The UTxO-HD overview file is still empty. It is difficult to write down something not too specific as for end users the only change will be (once LSM is in place) that less memory will be used. @nfrisby suggested writing it _as if_ we had finished the feature with LSM already.

**Comment by jasagredo** [comment]
This was already covered in the released version.
